### PR TITLE
feat: publish lockstep OCI container

### PR DESCRIPTION
**Pull request by mbeacom**
## What and why

Add a source-built OCI distribution for adrkit so the CLI, dual-era MCP server, governing-decisions Action, and queue Action can run without installing Node or Bun in the consuming repository.

The published `ghcr.io/mbeacom/adrkit` image follows the lockstep release rather than creating a drifting version channel. It is built for amd64/arm64 only after the coordinated Release workflow succeeds, pushed by digest, attested, then promoted to immutable `vX.Y.Z` and guarded moving tags. ADR-0032 records the release, recovery, provenance, and rollback contract.

The implementation also closes the deep-review findings: build-context secrets are excluded, dedicated targets contain only their executable, MCP examples use read-only mounts, both protocol eras are exercised, both Action boundaries are tested, CI is bounded, base indexes are digest-pinned, selector help is discoverable, and historical recovery cannot rewind moving tags or overwrite an immutable digest.

## Checklist

- [x] Commits are **DCO signed off**. No CLA is required.
- [x] ADR-0032 records and ratifies the new public distribution surface.
- [x] The schema is unchanged.
- [x] `packages/ci/src` and `@adrkit/core` are unchanged; the committed Action bundles remain byte-identical under Bun 1.3.14.
- [x] New behavior has contract and runtime smoke coverage, including observed failing build-context, help, and Action-boundary cases before correction.
- [x] `bun run typecheck`, `bun run build`, `bun test`, and `bun run lint` pass; Action workflows and the ADR corpus also lint cleanly.

## Notes for reviewers

- Only the all-in-one `adrkit` target is published. Dedicated targets remain local isolation/SBOM surfaces.
- The first GHCR image will be the first lockstep release containing ADR-0032; `v0.10.0` is intentionally not backfilled because that tagged tree predates the Containerfile.
- The first package publish requires a one-time human change to public GHCR visibility, documented in `docs/RELEASING.md`.

### CODE REVIEW

File: scripts/container-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+selectors:
+  cli | adr | adrkit              Run the adr CLI
+  mcp | adrkit-mcp                Run the stdio MCP server
+  ci | adrkit-ci                  Run the governing-decisions GitHub Action
+  queue-action | adrkit-queue-action
+                                  Run the managed ARB queue GitHub Action
+
+Any first argument not listed above is passed to the adr CLI.
+Run `adrkit-container cli --help` for CLI command help.
```
[the remark this answers is not in the document]
Comment (mbeacom) [reply scripts/container-entrypoint.sh]: Fixed in 707dbee. `help` is no longer intercepted, so `help <command>` reaches the CLI. Container help now explicitly lists the reserved forms (no args, `-h`, `--help`, `container-help`, and `--container-help`) and points to `cli --help`; README and contract coverage were updated to match.

> [!NOTE]
> Auto-replied by the [GitHub Copilot app](https://github.com/github/app).

File: .github/workflows/container-release.yml [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+          fi
+
+          releases=$(gh release list \
+            --repo "$GITHUB_REPOSITORY" \
+            --limit 100 \
+            --json isDraft,isPrerelease,tagName)
+          stable_tags=$(jq -r \
+            '.[] | select(.isDraft == false and .isPrerelease == false) | .tagName
+             | select(test("^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)$"))' \
+            <<<"$releases")
```
[the remark this answers is not in the document]
Comment (mbeacom) [reply .github/workflows/container-release.yml]: Fixed in 707dbee. Promotion now reads the complete release history with `gh api --paginate` (`per_page=100`) and filters stable lockstep tags on every page before computing the newest overall and per-major tags, so adapter/prerelease volume cannot truncate recovery history.

> [!NOTE]
> Auto-replied by the [GitHub Copilot app](https://github.com/github/app).

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -148,6 +148,93 @@ jobs:
...
+            'test -f /opt/adrkit/queue-action.js && test ! -e /opt/adrkit/adr.js && test ! -e /opt/adrkit/ci.js'
+      - name: Smoke test CLI and both MCP protocol eras without runtime network access
+        run: |
+          set -euo pipefail
+          test "$(docker image inspect --format '{{.Config.User}}' adrkit:ci)" = node
+          test "$(docker run --rm --read-only --network none adrkit-cli:ci --version)" = "$(jq -r .version package.json)"
+          docker run --rm --read-only --network none \
+            -v "$GITHUB_WORKSPACE:/workspace:ro" \
+            adrkit-cli:ci lint --dir docs/adr
+          node scripts/smoke-container.mjs adrkit-mcp:ci
```
[the remark this answers is not in the document]
Comment (mbeacom) [reply .github/workflows/ci.yml:193]: Fixed in 707dbee. CI now exercises CLI version/lint, both MCP protocol eras, the governing-decisions Action, and the queue Action through the published `adrkit:ci` selector layer. Dedicated targets remain separately built and checked for filesystem isolation.

> [!NOTE]
> Auto-replied by the [GitHub Copilot app](https://github.com/github/app).
